bug(`forge test`): compilation failed due to submodule resolution failure

### Component

Forge

### Have you ensured that all of these are up to date?

- [x] Foundry
- [x] Foundryup

### What version of Foundry are you on?

`b9bc92a471d7691e2db704c4c036d3499b62401f`

### What version of Foundryup are you on?

`1.5.0`

### What command(s) is the bug in?

`forge test`

### Operating System

Linux

### Describe the bug

My nightly snekmate tests started to fail last night: https://github.com/pcaversaccio/snekmate/actions/runs/20611814018/job/59208086609#step:15:1

```console
Run forge test
Compiler run failed:
Error (6275): Source "@openzeppelin/contracts/utils/Address.sol" not found: File not found. Searched the following locations: "/home/runner/work/snekmate/snekmate", "/home/runner/work/snekmate/snekmate/src/snekmate", "/home/runner/work/snekmate/snekmate/src/snekmate/auth", "/home/runner/work/snekmate/snekmate/src/snekmate/extensions", "/home/runner/work/snekmate/snekmate/src/snekmate/governance", "/home/runner/work/snekmate/snekmate/src/snekmate/tokens", "/home/runner/work/snekmate/snekmate/src/snekmate/tokens/interfaces", "/home/runner/work/snekmate/snekmate/src/snekmate/utils".
ParserError: Source "@openzeppelin/contracts/utils/Address.sol" not found: File not found. Searched the following locations: "/home/runner/work/snekmate/snekmate", "/home/runner/work/snekmate/snekmate/src/snekmate", "/home/runner/work/snekmate/snekmate/src/snekmate/auth", "/home/runner/work/snekmate/snekmate/src/snekmate/extensions", "/home/runner/work/snekmate/snekmate/src/snekmate/governance", "/home/runner/work/snekmate/snekmate/src/snekmate/tokens", "/home/runner/work/snekmate/snekmate/src/snekmate/tokens/interfaces", "/home/runner/work/snekmate/snekmate/src/snekmate/utils".
 --> lib/properties/contracts/ERC721/external/util/ERC721ExternalTestBase.sol:7:1:
  |
7 | import "@openzeppelin/contracts/utils/Address.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error (6275): Source "@openzeppelin/contracts/token/ERC721/utils/ERC721Holder.sol" not found: File not found. Searched the following locations: "/home/runner/work/snekmate/snekmate", "/home/runner/work/snekmate/snekmate/src/snekmate", "/home/runner/work/snekmate/snekmate/src/snekmate/auth", "/home/runner/work/snekmate/snekmate/src/snekmate/extensions", "/home/runner/work/snekmate/snekmate/src/snekmate/governance", "/home/runner/work/snekmate/snekmate/src/snekmate/tokens", "/home/runner/work/snekmate/snekmate/src/snekmate/tokens/interfaces", "/home/runner/work/snekmate/snekmate/src/snekmate/utils".
ParserError: Source "@openzeppelin/contracts/token/ERC721/utils/ERC721Holder.sol" not found: File not found. Searched the following locations: "/home/runner/work/snekmate/snekmate", "/home/runner/work/snekmate/snekmate/src/snekmate", "/home/runner/work/snekmate/snekmate/src/snekmate/auth", "/home/runner/work/snekmate/snekmate/src/snekmate/extensions", "/home/runner/work/snekmate/snekmate/src/snekmate/governance", "/home/runner/work/snekmate/snekmate/src/snekmate/tokens", "/home/runner/work/snekmate/snekmate/src/snekmate/tokens/interfaces", "/home/runner/work/snekmate/snekmate/src/snekmate/utils".
 --> lib/properties/contracts/ERC721/external/util/MockReceiver.sol:4:1:
  |
4 | import "@openzeppelin/contracts/token/ERC721/utils/ERC721Holder.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error (6275): Source "@openzeppelin/contracts/token/ERC721/IERC721.sol" not found: File not found. Searched the following locations: "/home/runner/work/snekmate/snekmate", "/home/runner/work/snekmate/snekmate/src/snekmate", "/home/runner/work/snekmate/snekmate/src/snekmate/auth", "/home/runner/work/snekmate/snekmate/src/snekmate/extensions", "/home/runner/work/snekmate/snekmate/src/snekmate/governance", "/home/runner/work/snekmate/snekmate/src/snekmate/tokens", "/home/runner/work/snekmate/snekmate/src/snekmate/tokens/interfaces", "/home/runner/work/snekmate/snekmate/src/snekmate/utils".
ParserError: Source "@openzeppelin/contracts/token/ERC721/IERC721.sol" not found: File not found. Searched the following locations: "/home/runner/work/snekmate/snekmate", "/home/runner/work/snekmate/snekmate/src/snekmate", "/home/runner/work/snekmate/snekmate/src/snekmate/auth", "/home/runner/work/snekmate/snekmate/src/snekmate/extensions", "/home/runner/work/snekmate/snekmate/src/snekmate/governance", "/home/runner/work/snekmate/snekmate/src/snekmate/tokens", "/home/runner/work/snekmate/snekmate/src/snekmate/tokens/interfaces", "/home/runner/work/snekmate/snekmate/src/snekmate/utils".
 --> lib/properties/contracts/ERC721/util/IERC721Internal.sol:4:1:
  |
Error: Compilation failed
4 | import "@openzeppelin/contracts/token/ERC721/IERC721.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error (6275): Source "@openzeppelin/contracts/token/ERC721/extensions/IERC721Enumerable.sol" not found: File not found. Searched the following locations: "/home/runner/work/snekmate/snekmate", "/home/runner/work/snekmate/snekmate/src/snekmate", "/home/runner/work/snekmate/snekmate/src/snekmate/auth", "/home/runner/work/snekmate/snekmate/src/snekmate/extensions", "/home/runner/work/snekmate/snekmate/src/snekmate/governance", "/home/runner/work/snekmate/snekmate/src/snekmate/tokens", "/home/runner/work/snekmate/snekmate/src/snekmate/tokens/interfaces", "/home/runner/work/snekmate/snekmate/src/snekmate/utils".
ParserError: Source "@openzeppelin/contracts/token/ERC721/extensions/IERC721Enumerable.sol" not found: File not found. Searched the following locations: "/home/runner/work/snekmate/snekmate", "/home/runner/work/snekmate/snekmate/src/snekmate", "/home/runner/work/snekmate/snekmate/src/snekmate/auth", "/home/runner/work/snekmate/snekmate/src/snekmate/extensions", "/home/runner/work/snekmate/snekmate/src/snekmate/governance", "/home/runner/work/snekmate/snekmate/src/snekmate/tokens", "/home/runner/work/snekmate/snekmate/src/snekmate/tokens/interfaces", "/home/runner/work/snekmate/snekmate/src/snekmate/utils".
 --> lib/properties/contracts/ERC721/util/IERC721Internal.sol:5:1:
  |
5 | import "@openzeppelin/contracts/token/ERC721/extensions/IERC721Enumerable.sol";
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: Process completed with exit code 1.
```

One day before everything was smooth: https://github.com/pcaversaccio/snekmate/actions/runs/20588649005

I haven't committed anything to snekmate for over a week and the remapping config hasn't been changed for months.

--- a/crates/config/src/lib.rs
+++ b/crates/config/src/lib.rs
@@ -774,40 +774,28 @@ impl Config {
     }
 
     fn from_figment(figment: Figment) -> Result<Self, ExtractConfigError> {
-        // Helper to add a profile only if it's not already present
-        fn add_profile(profiles: &mut Vec<Profile>, profile: &Profile) {
-            if !profiles.contains(profile) {
-                profiles.push(profile.clone());
-            }
-        }
-
         let mut config = figment.extract::<Self>().map_err(ExtractConfigError::new)?;
-        let active_profile = figment.profile().clone();
+        config.profile = figment.profile().clone();
 
-        // Collect profiles from the profile section.
         // The `"profile"` profile contains all the profiles as keys.
+        let mut add_profile = |profile: &Profile| {
+            if !config.profiles.contains(profile) {
+                config.profiles.push(profile.clone());
+            }
+        };
         let figment = figment.select(Self::PROFILE_SECTION);
         if let Ok(data) = figment.data()
             && let Some(profiles) = data.get(&Profile::new(Self::PROFILE_SECTION))
         {
             for profile in profiles.keys() {
-                add_profile(&mut config.profiles, &Profile::new(profile));
+                add_profile(&Profile::new(profile));
             }
         }
-
-        // Always include the default profile
-        add_profile(&mut config.profiles, &Self::DEFAULT_PROFILE);
-
-        // Ensure the active profile exists.
-        if config.profiles.contains(&active_profile) {
-            config.profile = active_profile;
-        } else {
-            return Err(ExtractConfigError::new(Error::from(format!(
-                "Profile {active_profile} does not exist"
-            ))));
-        }
+        add_profile(&Self::DEFAULT_PROFILE);
+        add_profile(&config.profile);
 
         config.normalize_optimizer_settings();
+
         Ok(config)
     }
 
@@ -6422,24 +6410,6 @@ mod tests {
             assert!(cfg.warnings.iter().any(
                 |w| matches!(w, crate::Warning::UnknownKey { key, .. } if key == "unknown_key_xyz")
             ));
-            Ok(())
-        });
-    }
-
-    #[test]
-    fn fails_on_unknown_profile() {
-        figment::Jail::expect_with(|jail| {
-            jail.create_file(
-                "foundry.toml",
-                r#"
-                [profile.default]
-                "#,
-            )?;
-
-            jail.set_env("FOUNDRY_PROFILE", "foo");
-            let err = Config::load().expect_err("expected unknown profile to fail");
-            assert!(err.to_string().contains("Profile foo does not exist"));
-
             Ok(())
         });
     }

